### ISSUE TITLE
Stacked LSTM - Documentation Example has Inconsistent Labels

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#9443]
I am running examples in the [Guide to the Sequential model documentation](https://keras.io/getting-started/sequential-model-guide/#examples). Specifically, I am running **Stacked LSTMs for sequence classification**

    data_dim = 16
    timesteps = 8
    num_classes = 10

    model = Sequential()
    model.add(LSTM(32, return_sequences=True,
               input_shape=(timesteps, data_dim)))
    model.add(LSTM(32, return_sequences=True))
    model.add(LSTM(32))  # return a single vector of dimension 32
    model.add(Dense(10, activation='softmax'))

    model.compile(loss='categorical_crossentropy',
              optimizer='rmsprop',
              metrics=['accuracy'])

    # Generate dummy data
    x_train = np.random.random((1000, timesteps, data_dim))
    y_train = np.random.random((1000, num_classes))
    x_val = np.random.random((100, timesteps, data_dim))
    y_val = np.random.random((100, num_classes))

    model.fit(x_train, y_train,
          batch_size=64, epochs=5,
          validation_data=(x_val, y_val))

This model compiles and runs. My question is with regard to `y_train` and `y_val`. The [loss documentation](https://keras.io/losses/#categorical_crossentropy) states that:

> Note: when using the categorical_crossentropy loss, your targets should be in categorical format (e.g. if you have 10 classes, the target for each sample should be a 10-dimensional vector that is all-zeros except for a 1 at the index corresponding to the class of the sample).

Shouldn't `y_train` and `y_val` be 1-hot encoded? If not, does Keras use the combination of the `categorical_crossentropy` loss and the `softmax` activation to infer that a vector `y` of floats in `[0.0, 1.0)` with dim `(1, num_classes)` has the label `argmax(y)`?
